metaflow/plugins/argo/argo_workflows.py

```diff
@@ -0,0 +1,1304 @@
+        #                            -- Array<DAGTask> |
+        #                                       |      |
+        #                                        -- Template
+        #
+        # Steps in FlowSpec are represented as DAGTasks.
+        # A DAGTask can reference to -
+        #     a ContainerTemplate (for linear steps..) or
+        #     another DAGTemplate (for nested foreaches).
+        #
+        # While we could have very well inlined container templates inside a DAGTask,
+        # unfortunately Argo variable substitution ({{pod.name}}) doesn't work as
+        # expected within DAGTasks
+        # (https://github.com/argoproj/argo-workflows/issues/7432) and we are forced to
+        # generate container templates at the top level (in WorkflowSpec) and maintain
+        # references to them within the DAGTask.
+
+        labels = {
+            "app": "metaflow",
+            "metaflow/argo_workflow_template_name": self.name,
+            # "metaflow/owner": self.username,
+            "metaflow/user": "argo-workflows",
+        }
+        # TODO: Identify all the attributes that should be published as labels.
+
+        annotations = {
+            "metaflow/production_token": self.production_token,
+            "metaflow/owner": self.username,
+            "metaflow/user": "argo-workflows",
+            "metaflow/flow_name": self.flow.name,
+        }
+        if current.get("project_name"):
+            annotations.update(
+                {
+                    "metaflow/project_name": current.project_name,
+                    "metaflow/branch_name": current.branch_name,
+                    "metaflow/project_flow_name": current.project_flow_name,
+                }
+            )
+
+        return (
+            WorkflowTemplate()
+            .metadata(
+                # Workflow Template metadata.
+                ObjectMeta()
+                .name(self.name)
+                # Argo currently only supports Workflow-level namespaces. When v3.4.0
+                # is released, we should be able to support multi-namespace /
+                # multi-cluster scheduling.
+                .namespace(KUBERNETES_NAMESPACE)
+                .labels(labels)
+                .annotations(annotations)
+            )
+            .spec(
+                WorkflowSpec()
+                # Set overall workflow timeout.
+                .active_deadline_seconds(self.workflow_timeout)
+                # Allow Argo to archive all workflow execution logs
+                .archive_logs()
+                # Automount service tokens for now - https://github.com/kubernetes/kubernetes/issues/16779#issuecomment-159656641
+                # TODO: Service account names are currently set in the templates. We
+                #       can specify the default service account name here to reduce
+                #       the size of the generated YAML by a tiny bit.
+                .automount_service_account_token()
```

Comment:
Why does it need a SA token? Does the pod itself need to call any K8S API?

Reply:
Not right now - but to support `@parallel` we would need to.